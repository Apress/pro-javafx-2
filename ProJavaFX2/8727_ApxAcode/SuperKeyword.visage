/*
 * @test
 * @run
 */

class Hamburger {
  function whatsInIt() {
    "Beef patties"
  }
}

class HamburgerWithCheese extends Hamburger {
  override function whatsInIt() {
    "{super.whatsInIt()} and cheese"
  }
}

var hamburger = Hamburger {};
var cheeseburger = HamburgerWithCheese {};
println("hamburger.whatsInIt() = {hamburger.whatsInIt()}.");
println("cheeseburger.whatsInit() = {cheeseburger.whatsInIt()}."); 
