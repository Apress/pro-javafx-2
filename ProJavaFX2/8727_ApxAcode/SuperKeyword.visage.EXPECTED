hamburger.whatsInIt() = Beef patties.
cheeseburger.whatsInit() = Beef patties and cheese.
